src/System.Drawing.Common/src/mono/System.Drawing.Drawing2D/GraphicsPath.cs

```diff
@@ -0,0 +1,896 @@
+//
+// System.Drawing.Drawing2D.GraphicsPath.cs
+//
+// Authors:
+//
+//   Miguel de Icaza (miguel@ximian.com)
+//   Duncan Mak (duncan@ximian.com)
+//   Jordi Mas i Hernandez (jordi@ximian.com)
+//   Ravindra (rkumar@novell.com)
+//   Sebastien Pouliot  <sebastien@ximian.com>
+//
+// Copyright (C) 2004,2006-2007 Novell, Inc (http://www.novell.com)
+//
+// Permission is hereby granted, free of charge, to any person obtaining
+// a copy of this software and associated documentation files (the
+// "Software"), to deal in the Software without restriction, including
+// without limitation the rights to use, copy, modify, merge, publish,
+// distribute, sublicense, and/or sell copies of the Software, and to
+// permit persons to whom the Software is furnished to do so, subject to
+// the following conditions:
+// 
+// The above copyright notice and this permission notice shall be
+// included in all copies or substantial portions of the Software.
+// 
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+// LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+// WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+//
+
+using System.ComponentModel;
+using System.Runtime.InteropServices;
+
+namespace System.Drawing.Drawing2D
+{
+	public sealed class GraphicsPath : MarshalByRefObject, ICloneable, IDisposable
+	{
+		// 1/4 is the FlatnessDefault as defined in GdiPlusEnums.h
+		private const float FlatnessDefault = 1.0f / 4.0f;
+
+		internal IntPtr nativePath = IntPtr.Zero;
+
+		GraphicsPath (IntPtr ptr)
+		{
+			nativePath = ptr;
+		}
+
+		public GraphicsPath ()
+		{
+                        Status status = SafeNativeMethods.Gdip.GdipCreatePath (FillMode.Alternate, out nativePath);
+                        SafeNativeMethods.Gdip.CheckStatus (status);
```

Comment:
I'm unclear on something: if libgdiplus is meant to provide the same surface area and behavior as gdiplus, why does anything about the interop layer differ between platforms? i.e. why does this file exist at all?